gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -205,39 +208,43 @@ def self._on_method_added(hook_mod, method_name, is_singleton_method: false)
     # This wrapper is very slow, so it will subsequently re-wrap with a much faster wrapper
     # (or unwrap back to the original method).
```

Comment:
This diff is extremely hard to read -- 
All I've done is:

1. wrap `T::Private::ClassUtils.replace_method(mod, method_name) do |*args, &blk|` in a `syncrhonize`
2. wrap `method_sig = T::Private::Methods.run_sig_block_for_method(new_method)` in the inner block in a `syncrhonize`